client_service_factory.js

```diff
@@ -0,0 +1,34 @@
+const {ClientService} = require('./client_service');
+const {IllegalArgumentError} = require('./illegal_argument_error');
+
+/**
+ * @class
+ */
+class ClientServiceFactory {
+  /**
+     * @constructor
+     * @param {LedgerClient} ledgerClient The LedgerClient object to inject
+     * @param {Protobuf} protobuf The protobuf message object to inject
+     */
+  constructor(ledgerClient, protobuf) {
+    if (!ledgerClient || !protobuf) {
+      throw new IllegalArgumentError(
+          'Can not initiate ClientServiceFactory with undefiend arguments'
+      );
+    }
+    this.ledgerClient = ledgerClient;
+    this.protobuf = protobuf;
+  }
+
+  /**
+     * @param {Object} properties JSON Object used for setting client properties
+     * @return {ClientService}
+     */
+  create(properties) {
```

Comment:
It this going to be like this ?
```
var factory = new ClientServiceFactory(ledger, protobuf);
var client = factory.create(properties);
```

If so, I'm not really sure about the benefit of having the factory because we can just write it as follows.
```
var client = new ClientService(ledger, protobuf, properties);
```

Or, `ledger` and `protobuf` are injected by some DI module ?
I might be completely misunderstanding something. 
It would be nice if you can clarify it more.

Reply:
I didn't design it well.
I was planning to use the same name `ClientService` so that I created a factory to generate its instance but it makes the implementation confusing. 

I think we could just rename ClientService to ClientServiceBase(ledgerClient, protobufMessage, clientProperties), and in node/web SDK we just implement ClientService class like
```
 */
class ClientService extends ClientServiceBase {
  constructor(properties) {
   const ledgerClient = ...; // it is generated differently between node.js and browser runtime
   const protobuf = ...; // it is generated differently between node.js and browser runtime
    super(ledgerClient, protobuf, properties);
  }
}
```